src/transformers/dynamic_module_utils.py

```diff
@@ -143,8 +143,40 @@ def get_class_in_module(class_name, module_path):
     """
     Import a module on the cache directory for modules and extract a class from it.
     """
+    module_dir = Path(HF_MODULES_CACHE) / os.path.dirname(module_path)
+    module_dir_backup_temp = str(module_dir) + "_backup_temp"
+    # make sure it doesn't exist yet
+    if os.path.isdir(module_dir_backup_temp):
+        shutil.rmtree(module_dir_backup_temp)
+    # copy to a temporary directory
+    shutil.copytree(module_dir, module_dir_backup_temp)
+
+    # remove `configuration.py`: this is necessary when we try to import modeling module, or other tokenizer/processor
+    # modules, while configuration module has been imported previously.
+    # TODO: This is only a simple heuristic. In general, we might need to consider any dynamic module that has been
+    #       imported. However, we don't have this information so far.
+    if os.path.isfile(f"{module_dir}/configuration.py"):
+        os.remove(f"{module_dir}/configuration.py")
+    # This has to be deleted too!
+    if os.path.isdir(f"{module_dir}/__pycache__"):
+        shutil.rmtree(f"{module_dir}/__pycache__")
```

Comment:
It turns out that we have to hide `configuration.py` and `__pycache__` while importing the modeling module.

Reply:
I tried to hide all files (other than the file of the target module to import), but it turns out that we also have a case
```python
from .util import ImportLayer
```
in the modeling file, see [hf-internal-testing/test_dynamic_model_with_util](https://huggingface.co/hf-internal-testing/test_dynamic_model_with_util/blob/main/modeling.py).

Unlike the config module being imported before the modeling module (so we don't need the file anymore), `util` is not imported at this moment, and its file should be in the directory.

Reply:
still not fully 100% working - hiding all other module files work, but can't do that as mentioned in the above comment.
Need to look deeper.